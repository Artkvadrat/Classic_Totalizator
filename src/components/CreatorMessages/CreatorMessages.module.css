.form-message {
    margin: 0;
    padding: 0;
    position: relative;
    width: 270px;
    margin-bottom: 30px;
    
}

.textarea-message {
    resize: none;
    background-color: transparent;
    outline: none;
    padding: 5px;
    width: 270px;
    height: 70px;
    border: 1px solid #F2EDED;

}
.create-message {
    background-color: #1890ff;
    color: #fff;
    outline: none;
    position: absolute;
    bottom: -26px;
    z-index: 5;
    right: 0;
    border: none;
    cursor: pointer;
}