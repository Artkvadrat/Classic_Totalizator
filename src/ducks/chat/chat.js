import HTTPService from '../../services/HTTPService/HTTPService';

export const REQUESTED = 'chat/requested';
export const RECEIVED = 'chat/received';
export const CREATED = 'event/create';
export const DELETED = 'event/delete';

const requested = () => ({
  type: REQUESTED
});

const received = (data) => ({
  type: RECEIVED,
  payload: data
});

const deleted = () => ({
  type: DELETED
});

export const deleteMessage = (id) => (dispatch) =>
  HTTPService.request({
    method: 'DELETE',
    path: `/api/Chat/${id}`
  }).then(dispatch(deleted()));

export const getMessages = () => (dispatch) => {
  dispatch(requested());

  return HTTPService.request({ path: '/api/Chat' }).then((data) => {
    dispatch(received(data));
  });
};

export const createMessage = (data) => (dispatch) =>
  HTTPService.request({
    method: 'POST',
    path: '/api/Events',
    body: { text: data }
  });

const initialState = {
  isLoading: false,
  messages: []
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case REQUESTED:
      return {
        ...state,
        isLoading: true
      };

    case RECEIVED:
      return {
        ...state,
        isLoading: false,
        messages: [...action.payload.messages]
      };

    // case FINISH:
    //   return {
    //     ...state,
    //     eventsData: state.eventsData.map((event) => {
    //       if (event.id === action.payload.id) {
    //         return { ...event, isEnded: true };
    //       }

    //       return event;
    //     })
    //   };

    default:
      return state;
  }
};

export default reducer;
