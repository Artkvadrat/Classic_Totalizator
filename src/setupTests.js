import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

enzyme.configure({ adapter: new Adapter() });

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn()
};
window.localStorage = localStorageMock;
