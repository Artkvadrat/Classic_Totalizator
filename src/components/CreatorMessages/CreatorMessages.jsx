import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { createMessage } from '../../ducks/chat/chat';
import styles from './CreatorMessages.module.css';

const CreatorMessages = () => {
  const dispatch = useDispatch();
  const [message, setMessage] = useState('');

  const handleChange = (event) => {
    setMessage(() => event.target.value);
  };

  const submit = (e) => {
    e.preventDefault();

    if (message !== '') {
      // dispatch(createMessage(message));
      setMessage(() => '');
    }
  };

  return (
    <form className={styles['form-message']} onSubmit={submit}>
      <textarea
        className={styles['textarea-message']}
        value={message}
        onChange={handleChange}
      />
      <input
        className={styles['create-message']}
        type="submit"
        value="Add message"
      />
    </form>
  );
};
export default CreatorMessages;
