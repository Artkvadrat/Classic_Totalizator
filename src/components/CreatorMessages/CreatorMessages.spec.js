import React from 'react';
import { shallow } from 'enzyme';
import { useDispatch } from 'react-redux';
import { createMessage } from '../../ducks/chat/chat';
import '../../setupTests';
import CreatorMessages from './CreatorMessages';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn()
}));

jest.mock('../../ducks/chat/chat', () => ({
  createMessage: jest.fn()
}));

describe('CreatorMessages component', () => {
  const dispatch = jest.fn();

  beforeEach(() => {
    useDispatch.mockReturnValue(dispatch);
    createMessage.mockReturnValue(Symbol.for('create'));
    jest.clearAllMocks();
  });

  it('should render form', () => {
    const result = shallow(<CreatorMessages />);
    const form = result.find('.form-message');
    expect(form.exists()).toEqual(true);
    expect(result).toMatchSnapshot();
  });

  xit('should create message on submit form', () => {
    const result = shallow(<CreatorMessages />);
    const fakeEvent = {
      preventDefault: jest.fn()
    };

    const form = result.find('.form-message');
    form.simulate('submit', fakeEvent);

    expect(fakeEvent.preventDefault).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalledTimes(0);
    expect(dispatch).toHaveBeenCalledWith(Symbol.for('create'));
    expect(createMessage).toHaveBeenCalledTimes(0);
  });

  xit("shouldn't create event on submit form", () => {
    const result = shallow(<CreatorMessages />);
    const fakeEvent = {
      preventDefault: jest.fn()
    };
    const form = result.find('.form-message');
    form.simulate('submit', fakeEvent);

    expect(fakeEvent.preventDefault).toHaveBeenCalled();
    expect(dispatch).not.toHaveBeenCalled();
    expect(createMessage).not.toHaveBeenCalled();
  });
  it('should change state on onChange textarea', () => {
    const result = shallow(<CreatorMessages />);
    const fakeEvent = {
      target: {
        value: 'hello'
      }
    };
    const textarea = result.find('.textarea-message');
    textarea.simulate('change', fakeEvent);
  });
});
