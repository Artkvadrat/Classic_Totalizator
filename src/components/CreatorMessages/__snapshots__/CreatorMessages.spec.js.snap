// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorMessages component should render form 1`] = `
<form
  className="form-message"
  onSubmit={[Function]}
>
  <textarea
    className="textarea-message"
    onChange={[Function]}
    value=""
  />
  <input
    className="create-message"
    type="submit"
    value="Add message"
  />
</form>
`;
